File: ruby/grains/grains.rb
class Grains
  def self.square(n)
    raise ArgumentError if n < 1 || n > 64
    2 ** (n - 1)
  end

  def self.total
    (0...64).map{|n| 2 ** n}.inject(&:+)
  end
end

module BookKeeping
  VERSION = 1
end

File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n) =>
        n < 1 || n > 64 
            ? throw new ArgumentOutOfRangeException() 
            : (ulong)Math.Pow(2, n - 1);

    public static ulong Total()
    {
        var total = 0ul;
        for (int i = 1; i <= 64; i++)
        {
            total += Square(i);
        }
        return total;
    }
}